src/routes/settings-knowledge-base.js

```diff
@@ -99,39 +143,59 @@ class SettingsKnowledgeBaseRoute extends Component {
     }
   };
 
+  deleteKBCredentials = kbID => {
+    this.setState({
+      currentKBName: this.getCurrentKBName(),
+    });
+    this.props.deleteKBCredentials(kbID);
+  };
+
   render() {
     const { kbCredentials } = this.props;
 
     return (
-      <FormattedMessage id="ui-eholdings.label.settings">
-        {pageLabel => (
-          <FormattedMessage id="ui-eholdings.settings.kb">
-            {recordLabel => (
-              <TitleManager
-                page={pageLabel}
-                record={recordLabel}
-              >
-                <View
-                  kbCredentials={kbCredentials}
-                  config={this.getCurrentConfig()}
-                  onSubmit={this.updateConfig}
-                  isCreateMode={this.state.isCreateMode}
-                />
-              </TitleManager>
-            )}
-          </FormattedMessage>
-        )}
-      </FormattedMessage>
+      <CalloutContext.Consumer>
+        {({ sendCallout }) => {
+          this.sendDeleteSuccessCallout = sendCallout;
+
+          return (
+            <FormattedMessage id="ui-eholdings.label.settings">
+              {pageLabel => (
+                <FormattedMessage id="ui-eholdings.settings.kb">
+                  {recordLabel => (
+                    <TitleManager
+                      page={pageLabel}
```

Comment:
Can we use `intl.formatMessage` here instead? It'll simplify the structure a lot